esi_leap/api/controllers/v1/utils.py

```diff
@@ -0,0 +1,84 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import oslo_policy
+
+from esi_leap.common import exception
+from esi_leap.common import policy
+
+
+def contract_get_all_authorize_filters(cdict, r_project_id,
+                                       start_time=None, end_time=None,
+                                       status=None, offer_uuid=None,
+                                       project_id=None, view=None, owner=None):
+
+    policy.authorize('esi_leap:contract:get', cdict, cdict)
+
+    possible_filters = {
+        'status': status,
+        'offer_uuid': offer_uuid,
+        'start_time': start_time,
+        'end_time': end_time,
+    }
+
+    try:
+        policy.authorize('is_admin', cdict, cdict)
+        is_admin = True
+
+    except oslo_policy.policy.PolicyNotAuthorized:
+        is_admin = False
+
+    if view == 'all':
+        policy.authorize('esi_leap:contract:get_all_all_view', cdict, cdict)
+        possible_filters['owner'] = owner
+        possible_filters['project_id'] = project_id
+    elif view == 'owned':
+
+        policy.authorize('esi_leap:contract:get_all_owned_view', cdict, cdict)
+
+        if owner is None:
+            owner = r_project_id
+        else:
+            if not is_admin:
+                if r_project_id != owner:
+                    raise exception.ProjectNoPermission(project_id=project_id)
+
+        possible_filters['owner'] = owner
+        possible_filters['project_id'] = project_id
+    else:
+
+        policy.authorize('esi_leap:contract:get_all_no_view', cdict, cdict)
+
+        if owner:
+            raise exception.InvalidOwnerCommand(view=view, owner=owner)
+
+        if project_id is None:
+            project_id = r_project_id
+        else:
+            if not is_admin:
+                if project_id != r_project_id:
+                    raise exception.ProjectNoPermission(project_id=project_id)
```

Comment:
Ah, yep, this is something that could then be replaced by the new util function.